### DISCUSSION TITLE
Write to BytesIO and read BytesIO

### DISCUSSION THREAD

**Discussion by jommme** [mozman/ezdxf#1205] [Q&A]
I am currently writing to a BytesIO in the following way:
with BytesIO() as buffer:
                doc.write(buffer, fmt="bin")
                buffer.seek(0)
                binary_file = buffer.read()
                
I am looking or a way to open the binary_file with ezdxf at a later stage.
Currently I have tried:

ezdxf.read(BytesIO(binary_file))

But than i get the following error:
raise DXFStructureError(f'Invalid group code "{code}" at line {line}.')
ezdxf.lldxf.const.DXFStructureError: Invalid group code "b'AutoCAD Binary DXF\r\n'" at line 1.

Anyone know how I can solve this problem?

**Comment by mozman** [comment]
`ezdxf.read()` reads only ASCII-DXF from text-streams. Binary DXF can only be loaded from a file by `ezdxf.readfile()`. 

You can here learn how `ezdxf` loads binary-DXF files and implement your own loading function: 

https://github.com/mozman/ezdxf/blob/0f0ac03ddef1bd4db71dd70d42ab0c5f89a0c1ba/src/ezdxf/filemanagement.py#L138

```Python
  if is_binary_dxf_file(filename):
        with open(filename, "rb") as fp:
            data = fp.read()
            loader = binary_tags_loader(data, errors=errors)
            doc = Drawing.load(loader)
            doc.filename = filename
            return doc
```
BTW: Writing Binary DXF files is not faster than writing ASCII-DXF files and has some edge cases where the DXF file cannot be read by AutoCAD.
